src/ProductConstructionService/ProductConstructionService.ReproTool/ReproTool.cs

```diff
@@ -0,0 +1,321 @@
+        // Sync the VMR fork branch
+        await SyncForkAsync("dotnet", "dotnet", branch);
+        // Check if the user has the forked VMR in local DB
+        await AddRepositoryToBarIfMissingAsync(VmrForkUri);
+
+        var newBranch = await CreateTmpBranchAsync(VmrForkRepoName, branch);
+
+        // Fetch source mappings and source manifest files and replace the mapping for the repo we're testing on
+        logger.LogInformation("Updating source mappings and source manifest files in VMR fork to replace original product repo mapping with fork mapping");
+        await UpdateRemoteVmrForkFileAsync(newBranch.Value, productRepoUri, productRepoForkUri, SourceMappingsPath);
+        await UpdateRemoteVmrForkFileAsync(newBranch.Value, productRepoUri, productRepoForkUri, SourceManifestPath);
+
+        return newBranch;
+    }
+
+    private async Task DeleteGitHubBranchAsync(string repo, string branch) => await ghClient.Git.Reference.Delete(MaestroAuthTestOrgName, repo, $"heads/{branch}");
+
+    private async Task UpdateRemoteVmrForkFileAsync(string branch, string productRepoUri, string productRepoForkUri, string filePath)
+    {
+        // Fetch remote file and replace the product repo URI with the repo we're testing on        
+        var sourceMappingsFile = (await ghClient.Repository.Content.GetAllContentsByRef(
+            MaestroAuthTestOrgName,
+            VmrForkRepoName,
+            filePath,
+            branch)).FirstOrDefault() ??
+                throw new Exception($"Failed to find file {SourceMappingsPath} in {MaestroAuthTestOrgName}" +
+                    $"/{VmrForkRepoName} on branch {SourceMappingsPath}");
+
+        // Replace the product repo uri with the forked one
+        var updatedSourceMappings = sourceMappingsFile.Content.Replace(productRepoUri, productRepoForkUri);
+        UpdateFileRequest update = new(
+            $"Update {productRepoUri} source mapping",
+            updatedSourceMappings,
+            sourceMappingsFile.Sha,
+            branch);
+
+        await ghClient.Repository.Content.UpdateFile(
+            MaestroAuthTestOrgName,
+            VmrForkRepoName,
+            filePath,
+            update);
+    }
+
+    private async Task<AsyncDisposableValue<string>> PrepareProductRepoForkAsync(
+        string productRepoUri,
+        string productRepoForkUri,
+        string productRepoBranch)
+    {
+        (var name, var org) = GitRepoUrlParser.GetRepoNameAndOwner(productRepoUri);
+        // Check if the product repo fork already exists
+        var allRepos = await ghClient.Repository.GetAllForOrg(MaestroAuthTestOrgName);
+        
+        // If we already have a fork in maestro-auth-test, sync the branch we need with the source
+        if (allRepos.FirstOrDefault(repo => repo.HtmlUrl == productRepoForkUri) != null)
+        {
+            logger.LogInformation("Product repo fork {fork} already exists, syncing branch {branch} with source", productRepoForkUri, productRepoBranch);
```

Comment:
Move logging inside of the method

Reply:
I added logging o the method, but I think there's still value in keeping this here